packages/deployments/contracts/test/connext.spec.ts

```diff
@@ -1042,4 +1018,321 @@ describe("Connext", () => {
+        prepareReceipt.logs.find((l) => l.topics.includes(topics[0]))!,
+      );
+      message = (bridgeEvent!.args as any).message;
+      bridgedAmount = (bridgeEvent!.args as any).amount;
+
+      const originTmEvent = (await originTm.queryFilter(originTm.filters.XCalled())).find(
+        (a) => a.blockNumber === prepareReceipt.blockNumber,
+      );
+      nonce = (originTmEvent!.args as any).nonce;
+      transferId = (originTmEvent!.args as any).transferId;
+
+      reconciledTopics = destinationTm.filters.Reconciled().topics as string[];
+    });
+
+    it("should work with one router", async () => {
+      // Get pre-execute balances
+      const preExecute = await Promise.all([
+        destinationAdopted.balanceOf(user.address),
+        destinationTm.routerBalances(router.address, local.address),
+      ]);
+
+      // Fulfill with the router
+      const routersAmount = amount - 500;
+      const routers = [router.address];
+      const routerProportionalAmount = Math.floor(routersAmount / routers.length);
+      await destinationTm.connect(router).execute({
+        params,
+        nonce,
+        local: local.address,
+        amount: routersAmount,
+        feePercentage: constants.Zero,
+        relayerSignature: "0x",
+        routers,
+        originSender: user.address,
+      });
+
+      // Check balance of user + bridge
+      const postExecute = await Promise.all([
+        destinationAdopted.balanceOf(user.address),
+        destinationTm.routerBalances(router.address, local.address),
+      ]);
+      expect(postExecute[0]).to.be.eq(preExecute[0].add(routersAmount));
+      expect(postExecute[1]).to.be.eq(preExecute[1].sub(routerProportionalAmount));
+
+      // Reconcile via bridge
+      const preReconcile = await Promise.all([destinationTm.routerBalances(router.address, local.address)]);
+
+      const reconcile = await destinationBridge
+        .connect(admin)
+        .handle(originDomain, 0, addressToBytes32(originBridge.address), message);
+
+      const reconcileReceipt = await reconcile.wait();
+      const reconciledEvent = destinationTm.interface.parseLog(
+        reconcileReceipt.logs.find((l) => l.topics.includes(reconciledTopics[0]))!,
+      );
+      expect(reconciledEvent.args.transferId).eql(transferId);
+      expect(reconciledEvent.args.to).eql(user.address);
+      expect(reconciledEvent.args.localAsset).eql(local.address);
+      expect(reconciledEvent.args.localAmount).eql(bridgedAmount);
+      expect(reconciledEvent.args.executed).eql([routers, BigNumber.from(routersAmount)]);
+      expect(reconciledEvent.args.caller).eql(destinationBridge.address);
+
+      const routerReconciledAmount = Math.floor(amount / routers.length);
+      const postReconcile = await Promise.all([destinationTm.routerBalances(router.address, local.address)]);
+      expect(postReconcile[0]).to.be.eq(preReconcile[0].add(routerReconciledAmount));
+    });
+
+    it("should work with two routers", async () => {
```

Comment:
Seems like we could execute these tests in a `for` loop

Reply:
Made a refactor of the tests, duplicated code was removed, now it executes in a forEach loop https://github.com/connext/nxtp/pull/916/commits/0e74b944169e3883c7cf3c0985fde311d160b1e8